subsys/bluetooth/audio/has_client.c

```diff
@@ -219,28 +425,41 @@ static void handle_preset_changed(struct has_inst *inst, struct net_buf_simple *
 	}
 }
 
-static uint8_t control_point_notify_cb(struct bt_conn *conn,
-				       struct bt_gatt_subscribe_params *params, const void *data,
-				       uint16_t len)
+static void subscribed(struct bt_gatt_subscribe_params *params)
 {
-	struct has_inst *inst = CONTAINER_OF(params, struct has_inst, control_point_subscription);
-	const struct bt_has_cp_hdr *hdr;
-	struct net_buf_simple buf;
+	/* Clear the subscribe callback to not be called when subscription is removed */
+	params->subscribe = NULL;
```

Comment:
I understand why you do this, but this if effectively disregarding the behavior of the GATT layer, which also means that if that GATT layer change behavior, that behavior need to be copied here, since you are not relying on the API (callback). 

It's unlikely to be an issue, but it also does not feel like the right approach - Although on the other hand it's not much different than just ignoring the callback when it's being unsubscribed, so it is OK to keep as is.